### DISCUSSION TITLE
Why do publish examples declare queues and bind to exchange?

### DISCUSSION THREAD

**Discussion by space88man** [pika/pika#1478] [General]
I noticed that the publish examples declare queues and bind to the exchange.

From a minimal learning perspective: is there anything specific in the publish examples
that actually use(require) the queue? I don't seen anything consuming from the queue (not surprising
given that these are publish examples).

Is the some other reason why a queue is necessary (receive delivery confirmations? - random
thought ).

E.g. why not `start_publishing()` in `on_exchange_declareok()` — instead of exchange -> queue
-> start_publishing

From https://github.com/pika/pika/blob/main/examples/asynchronous_publisher_example.py

```python
    def on_exchange_declareok(self, _unused_frame, userdata):
        """Invoked by pika when RabbitMQ has finished the Exchange.Declare RPC
        command.

        :param pika.Frame.Method unused_frame: Exchange.DeclareOk response frame
        :param str|unicode userdata: Extra user data (exchange name)

        """
        LOGGER.info('Exchange declared: %s', userdata)
        self.setup_queue(self.QUEUE)

    def setup_queue(self, queue_name):
        """Setup the queue on RabbitMQ by invoking the Queue.Declare RPC
        command. When it is complete, the on_queue_declareok method will
        be invoked by pika.

        :param str|unicode queue_name: The name of the queue to declare.

        """
        LOGGER.info('Declaring queue %s', queue_name)
        self._channel.queue_declare(queue=queue_name,
                                    callback=self.on_queue_declareok)
```

```

**Comment by lukebakken** [comment]
The example uses delivery confirmations, so yes, a queue is required:

https://github.com/pika/pika/blob/main/examples/asynchronous_publisher_example.py#L228-L240
